keyboards/annepro2/protocol.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * (c) 2021 by Tomasz bla Fortuna
+ * License: GPLv2
+ *
+ * This file is shared with Shine firmware.
+ */
+
+#ifndef PROTOCOL_INCLUDED
+#define PROTOCOL_INCLUDED
+#include <inttypes.h>
+
+#define PROTOCOL_SD SD0
+
+enum {
+  /*
+   * Main -> LED
+   */
+  /* Basic config */
+  CMD_LED_ON = 0x01,
+  CMD_LED_OFF = 0x02,
+
+  CMD_LED_SET_PROFILE = 0x03,
+  CMD_LED_NEXT_PROFILE = 0x04,
+  CMD_LED_PREV_PROFILE = 0x05,
+
+  CMD_LED_NEXT_INTENSITY = 0x06,
+  CMD_LED_NEXT_ANIMATION_SPEED = 0x07,
+
+  /* Masks */
+  /* Override a key color, eg. capslock */
+  CMD_LED_MASK_SET_KEY = 0x10,
+  /* Override all keys in a row with configurable colors */
+  CMD_LED_MASK_SET_ROW = 0x11,
+
+  /* Override all keys with single color (eg. foreground color) */
+  CMD_LED_MASK_SET_MONO = 0x12,
+
+  /* Reactive / status */
+  CMD_LED_GET_STATUS = 0x20,
+  CMD_LED_KEY_BLINK = 0x21, /* TODO: Blink */
+  CMD_LED_KEY_DOWN = 0x22,
+  CMD_LED_KEY_UP = 0x23, /* TODO */
+  CMD_LED_IAP = 0x24,
+
+  /* LED -> Main */
+  /* TODO: Payload with data to send over HID */
+  CMD_LED_DEBUG = 0x40,
+
+  /* Number of profiles, current profile, on/off state,
+     reactive flag, brightness, errors */
+  CMD_LED_STATUS = 0x41,
+};
+
+/* 4 ROWS * 14 COLS * 4B (RGBX) = 224 */
+#define MAX_PAYLOAD_SIZE 230
+
+enum protoState {
```

Comment:
//! enum of all the FSA states used for the protocol (or something along the lines)

Reply:
This, and other smaller changes I've made as a fixup to previous commit altering protocols.